streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -369,6 +325,8 @@ public void prepareCommit() {
         switch (state()) {
             case RUNNING:
             case RESTORING:
+            case SUSPENDED:
+                maybeScheduleCheckpoint();
```

Comment:
Instead of "blindly" writing a checkpoint in `postCommit()`, we only do it if a checkpoint get's scheduled.